File: JAVA Language/Vectors and Matrix Medium Level/DotProduct.java
import java.util.Scanner;

public class DotProduct {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter the length of the vectors: ");
        int n = sc.nextInt();

        int[] v1 = new int[n];
        int[] v2 = new int[n];

        System.out.println("Enter the elements of the first vector:");
        for (int i = 0; i < n; i++) {
            v1[i] = sc.nextInt();
        }

        System.out.println("Enter the elements of the second vector:");
        for (int i = 0; i < n; i++) {
            v2[i] = sc.nextInt();
        }

        int dotProduct = 0;
        for (int i = 0; i < n; i++) {
            dotProduct += v1[i] * v2[i];
        }

        System.out.println("The dot product of the two vectors is: " + dotProduct);

        sc.close();
    }
}

File: PYTHON Language/Algorithms Medium Level/DotProduct.py
# Create a program that asks the user for two vectors of equal length.
# Then, find the dot product of the vectors and display it on the screen.

# Ask the user for the vectors
n = int(input("Enter the length of the vectors: "))
vector1 = input("Enter the first vector, separated by spaces: ")
vector1 = [float(num) for num in vector1.split()]
vector2 = input("Enter the second vector, separated by spaces: ")
vector2 = [float(num) for num in vector2.split()]

# Find the dot product of the vectors
dot_product = sum([vector1[i]*vector2[i] for i in range(n)])

# Print the dot product of the vectors
print("The dot product of the two vectors is:", dot_product)
